### ISSUE TITLE
Always missing one item in the prediction. 

### ISSUE THREAD

**Issue by phylliskaka** [matterport/Mask_RCNN#1231]
Hi All, 

I am recently training the Mask-RCNN on my dataset. The result is pretty good. But there is one issue bother me a lot. The object detection some time always missing one item when it really shouldnt.
![leaf12test](https://user-images.githubusercontent.com/34198600/51336514-55bf1900-1a4a-11e9-8fba-fa8612061b3e.png)
This prediction missed three leaves. And I also find same issue happened in the model trained with COCO dataset. 
![people1test](https://user-images.githubusercontent.com/34198600/51336635-aafb2a80-1a4a-11e9-9be2-619b495c6924.png)
If anyone faced the same issue, please tell me your thought. Really appreciate that

**Comment by phylliskaka** [comment]
Right now, I am think It might due to the iou of bounding box is too high, that why it filter the one with lower confidence level bounding box.

**Comment by chriscnapier** [comment]
I have the same problem as your person image. I have 5 people detected instead of 6. Your example is clearer than mine, and helps me to look further in the code. 
I am debugging (pdb) "/usr/local/lib/python3.7/dist-packages/tensorflow/python/keras/engine/training_arrays.model_iteration",
which returns one missing detection in results[0].
In my case I re-ran the MaskRCNN demo separately for each detection (by zeroing the pixels outside its bounding box) and then the program did detect the previously missing person, So its not a pixel size or scale issue or loss issue.
I believe the problem is that an object is missed when its bounding box is intruded on by two other bounding boxes !
